crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1233,75 +1231,23 @@ fn handle_with_item_comment<'a>(
     }
 }
 
-/// Handles trailing comments after the `as` keyword of a pattern match item:
-///
+/// Handles trailing comments between the class name and its arguments in:
 /// ```python
 /// case (
 ///     Pattern
 ///     # dangling
-///     (  # dangling
-///         # dangling
-///     )
+///     (...)
 /// ): ...
 /// ```
 fn handle_pattern_match_class_comment<'a>(
     comment: DecoratedComment<'a>,
     class: &'a ast::PatternMatchClass,
-    locator: &Locator,
 ) -> CommentPlacement<'a> {
-    // Find the open parentheses on the arguments.
-    let Some(left_paren) = SimpleTokenizer::starts_at(class.cls.end(), locator.contents())
-        .skip_trivia()
-        .find(|token| token.kind == SimpleTokenKind::LParen)
-    else {
-        return CommentPlacement::Default(comment);
-    };
-
-    // If the comment appears before the open parenthesis, it's dangling:
-    // ```python
-    // case (
-    //     Pattern
-    //     # dangling
-    //     (...)
-    // ): ...
-    // ```
-    if comment.end() < left_paren.start() {
-        return CommentPlacement::dangling(comment.enclosing_node(), comment);
-    }
-
-    let Some(first_item) = class
-        .patterns
-        .first()
-        .map(Ranged::start)
-        .or_else(|| class.kwd_attrs.first().map(Ranged::start))
-    else {
-        // If there are no items, then the comment must be dangling:
-        // ```python
-        // case (
-        //     Pattern(
-        //         # dangling
-        //     )
-        // ): ...
-        // ```
-        return CommentPlacement::dangling(comment.enclosing_node(), comment);
-    };
-
-    // If the comment appears before the first item or its parentheses, then it's dangling:
-    // ```python
-    // case (
-    //     Pattern(  # dangling
-    //         0,
-    //         0,
-    //     )
-    // ): ...
-    // ```
-    if comment.line_position().is_end_of_line() {
-        if comment.end() < first_item {
-            return CommentPlacement::dangling(comment.enclosing_node(), comment);
-        }
+    if class.cls.end() < comment.start() && comment.end() < class.arguments.start() {
+        CommentPlacement::dangling(comment.enclosing_node(), comment)
+    } else {
+        CommentPlacement::Default(comment)
     }
-
-    CommentPlacement::Default(comment)
```

Comment:
This gets dramatically simplified.

Reply:
...dramatically :laughing:

Reply:
Anything that deletes code is a "dramatic simplification" for me